### PR TITLE
Allow install custom functions refresh project

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
- Added both JS and TS branches for Custom Functions for Sept 2018 Preview.
- Removed May 2018 Preview branch so Yo Office will no longer pull in that branch

### CODE REVIEW

File: src/app/config/projectsJsonData.ts [hunk 1]
```diff
@@ -93,4 +93,36 @@ export default class projectsJsonData{
       }
       return undefined;
     }
+
+    getProjectTemplateBranchName(projectTypeKey: string, scriptType: string, branchIndex: number)
```
Comment (akrantz) [inline src/app/config/projectsJsonData.ts]: Why is there branchIndex?

File: src/app/index.ts [hunk 2]
```diff
@@ -263,7 +261,8 @@ module.exports = yo.extend({
         let projectRepo = jsonData.getProjectTemplateRepository(this.project.projectType, language == 'ts' ? _.toLower(typescript) : _.toLower(javascript));
```
Comment (akrantz) [inline src/app/index.ts]: This code could be written more simply. You can have one function that returns the object for a given projectType and language, and that object can have repository and branch properties. If (repository), then use git clone, and then if (branch) then add branch parameter to the git clone command.

Comment (TCourtneyOwen) [reply src/app/index.ts]: Fixed as suggested

File: src/app/config/projectsJsonData.ts (hunk 1 above)
Comment (TCourtneyOwen) [reply src/app/config/projectsJsonData.ts]: removed

File: src/app/config/projectProperties.json [hunk 3]
```diff
@@ -3,17 +3,23 @@
...
+                }     
         },
         "excel-functions": {
-            "displayname": "Excel Custom Functions Add-in project (Preview: Requires the Insider channel for Excel)",
+            "displayname": "Excel Custom Functions Add-in project (September 2018 Preview Refresh: Requires the Insider channel for Excel)",
             "templates": {
                 "javascript": {
-                    "repository": "https://github.com/OfficeDev/Excel-Custom-Functions"}
+                    "repository": "https://github.com/OfficeDev/Excel-Custom-Functions",
+                    "branches": [
```
Comment (akrantz) [inline src/app/config/projectProperties.json]: shouldn't there just be a single branch name, e.g. ` "branch": "preview-sept-2018-javascript", `

File: src/app/config/projectsJsonData.ts (hunk 1 above)
Comment (akrantz) [reply src/app/config/projectsJsonData.ts]: Do you still have changes to push?

File: src/app/config/projectsJsonData.ts [hunk 4]
```diff
@@ -94,35 +94,41 @@ export default class projectsJsonData{
...
       return undefined;
     }
+
+    getProjectRepoAndBranch(projectTypeKey: string, scriptType: string)
+    {
+      scriptType =  scriptType === 'ts' ? 'typescript' : 'javascript';
+
+      let repoBranchInfo = { repo: <string> null, branch: <string> null };
+      repoBranchInfo.repo = this.getProjectTemplateRepository(projectTypeKey, scriptType);
+      (repoBranchInfo.repo != undefined ) ? repoBranchInfo.branch = this.getProjectTemplateBranchName(projectTypeKey, scriptType) : repoBranchInfo.branch = undefined;
```
Comment (akrantz) [inline src/app/config/projectsJsonData.ts]: I think this line could be written more simply as:

` repoBranch.branch = (repoBranchInfo.repo) ? this.getProjectTemplateBranchName(projectTypeKey, scriptType) : undefined; `


This avoids the assignment inside the conditional statement which is confusing and error prone (as usually == is more often intended).

File: src/app/index.ts [hunk 5]
```diff
@@ -256,16 +256,15 @@ module.exports = yo.extend({
...
-        let jsonData = new projectsJsonData(this.templatePath()); 
+        let jsonData = new projectsJsonData(this.templatePath());
+        let projectRepoBranchInfo = jsonData.getProjectRepoAndBranch(this.project.projectType, language);
 
         this._projectCreationMessage();
         
         // Copy project template files from project repository (currently only custom functions has its own separate repo)
-        let projectRepo = jsonData.getProjectTemplateRepository(this.project.projectType, language == 'ts' ? _.toLower(typescript) : _.toLower(javascript));
-        if (projectRepo != "")
+        if (projectRepoBranchInfo.repo != "")
```
Comment (akrantz) [inline src/app/index.ts]: Just do: `if (projectRepoBranchInfo.repo)` which avoids any "falsy" value, including "", null, or undefined.

**Pull request merged**